receiver/splunkhecreceiver/splunk_to_logdata_test.go

```diff
@@ -123,6 +126,24 @@ func Test_SplunkHecToLogData(t *testing.T) {
 			}(),
 			wantErr: nil,
 		},
+		{
+			name: "nil_timestamp",
+			event: splunk.Event{
+				Time:       nil,
+				Host:       "localhost",
+				Source:     "mysource",
+				SourceType: "mysourcetype",
+				Index:      "myindex",
+				Event:      "value",
+				Fields: map[string]interface{}{
+					"foo": "bar",
+				},
+			},
+			output: func() pdata.ResourceLogsSlice {
+				return createLogsSlice("value")
+			}(),
+			wantErr: errors.New(nilTimestamp),
```

Comment:
Corresponding test for that primitive validation - again, I'm not sure if this is the way you'd like to go with the `splunkhecreceiver`